src/PhpSpec/ObjectBehavior.php

```diff
@@ -31,9 +31,90 @@
  * @method void beConstructedWith(...$arguments)
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments = array())
  * @method void beAnInstanceOf($class)
+ *
  * @method void shouldHaveType($type)
+ * @method void shouldNotHaveType($type)
+ * @method void shouldReturnAnInstanceOf($type)
```

Comment:
Adding them in Subject would not help at all, because your IDE will not know that ``$this->testedMethod()`` will return a Subject. We cannot add magic phpdoc for that. Knowing it in the IDE would require to build a phpspec integration in the IDE to know it. And then, such an integration could support matchers properly rather than relying on ``@method`` in the Subject phpdoc, ensuring support for non-core matchers